### PR TITLE
YARN-8262. get_executable in container-executor should provide meaningful error codes

### PR DESCRIPTION
**Pull request by susheel-gupta**
### Description of PR
Added meaningful error codes to get_executable in container-executor.
Earlier it was calling exit(-1) that makes it difficult to debug without stderr.

### How was this patch tested?
- Tested this patch by building a native profile hadoop on ycloud with image ycloud-base/base-centos7.8:0.1.0.0-13.
- Used this command to test the native testcases : **mvn test -Pnative -Dtest=allNative**

- First tested the behaviour of container executor exit code present already by calling out the exit(OUT_OF_MEMORY)  shown [here](https://drive.google.com/file/d/1NaipJ8NT4Y2yiP3wg_i5iMPVtpFCU7QA/view?usp=share_link)  , after this the error [found is like](https://drive.google.com/drive/folders/1Ax1LpbF2oLDE-ON3UOaV7d01-wAd7syw?usp=sharing) 
- Used the same method to test the error codes I added by providing a [patch](https://drive.google.com/file/d/1MHiOSM7Y-WJwPYZD3oZ0RHrdD60YzfZG/view?usp=share_link) on top of  hadoop, the testcases [failed like this as expected.](https://drive.google.com/file/d/1EutJl0yOoYxQMUwSxgOTHKmwxpIUn_5L/view?usp=share_link)
- After providing the correct patch the testcases passed. 
### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  45m 25s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  43m  1s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   1m 31s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 57s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  67m 33s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 41s |  |  the patch passed  |
| -1 :x: |  compile  |   0m 55s | [/patch-compile-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/1/artifact/out/patch-compile-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt) |  hadoop-yarn-server-nodemanager in the patch failed.  |
| -1 :x: |  cc  |   0m 55s | [/patch-compile-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/1/artifact/out/patch-compile-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt) |  hadoop-yarn-server-nodemanager in the patch failed.  |
| -1 :x: |  golang  |   0m 55s | [/patch-compile-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/1/artifact/out/patch-compile-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt) |  hadoop-yarn-server-nodemanager in the patch failed.  |
| -1 :x: |  javac  |   0m 55s | [/patch-compile-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/1/artifact/out/patch-compile-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt) |  hadoop-yarn-server-nodemanager in the patch failed.  |
| +1 :green_heart: |  blanks  |   0m  1s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 41s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  23m 18s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 53s | [/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/1/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt) |  hadoop-yarn-server-nodemanager in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 40s |  |  The patch does not generate ASF License warnings.  |
|  |   | 141m 20s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4949 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell detsecrets golang |
| uname | Linux 58186a4fa351 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / f13f2f1ce34083ff3df33f1dc056057572cf3b78 |
| Default Java | Red Hat, Inc.-1.8.0_345-b01 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/1/testReport/ |
| Max. process+thread count | 607 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/1/console |
| versions | git=2.9.5 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by szilard-nemeth** [comment]
Hi @susheel-gupta ,
Can you check the build errors, please?

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  38m  0s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  44m 51s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   1m 29s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 56s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  66m 31s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 24s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   1m 24s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   1m 24s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 24s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 44s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  19m 32s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  24m 15s |  |  hadoop-yarn-server-nodemanager in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 46s |  |  The patch does not generate ASF License warnings.  |
|  |   | 152m 56s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4949 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell detsecrets golang |
| uname | Linux 195db4d1f1aa 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / aa3dcae38808eeb26d2703140d579a9891f11dac |
| Default Java | Red Hat, Inc.-1.8.0_345-b01 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/2/testReport/ |
| Max. process+thread count | 727 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/2/console |
| versions | git=2.9.5 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  45m 55s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  43m  4s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   1m 29s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 56s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  67m 45s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 40s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 18s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   1m 18s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   1m 18s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 18s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  22m 26s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  23m 48s |  |  hadoop-yarn-server-nodemanager in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 42s |  |  The patch does not generate ASF License warnings.  |
|  |   | 164m 26s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4949 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell detsecrets golang |
| uname | Linux 2ef1ffdd9ac7 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 6b5f5c2c19edf5081ff6aac12dc03ca071f497c1 |
| Default Java | Red Hat, Inc.-1.8.0_345-b01 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/3/testReport/ |
| Max. process+thread count | 530 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/3/console |
| versions | git=2.9.5 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by K0K0V0K** [comment]
I think there should be at least one test in the TestLinuxContainerExecutor, what will check in case of an error the app will return with a not -1 error code.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  45m 56s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  50m 14s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   1m 31s |  |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   0m 49s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 55s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 49s |  |  trunk passed  |
| +1 :green_heart: |  spotbugs  |   1m 46s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  24m  4s |  |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  24m 25s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 20s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   1m 20s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   1m 20s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 20s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 44s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 33s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 39s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  23m 39s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  23m 43s |  |  hadoop-yarn-server-nodemanager in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 42s |  |  The patch does not generate ASF License warnings.  |
|  |   | 179m 18s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4949 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets cc golang |
| uname | Linux 93fdfb46d9fe 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 3f980cda6da4d6965786ab87312018f6bb0e257c |
| Default Java | Red Hat, Inc.-1.8.0_352-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/4/testReport/ |
| Max. process+thread count | 534 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/4/console |
| versions | git=2.9.5 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java [hunk 1]
```diff
@@ -175,8 +175,12 @@ public enum ExitCode {
     COULD_NOT_CREATE_WORK_DIRECTORIES(35),
     COULD_NOT_CREATE_APP_LOG_DIRECTORIES(36),
     COULD_NOT_CREATE_TMP_DIRECTORIES(37),
-    ERROR_CREATE_CONTAINER_DIRECTORIES_ARGUMENTS(38);
-
+    ERROR_CREATE_CONTAINER_DIRECTORIES_ARGUMENTS(38),
+    CANT_GET_EXECUTABLE_NAME_FROM_READLINK(80),
+    TOO_LONG_EXECUTOR_PATH(81),
+    CANT_GET_EXECUTABLE_NAME_FROM_KERNEL(82),
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java]: Replace CANT with CANNOT

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java [hunk 2]
```diff
@@ -175,8 +175,12 @@ public enum ExitCode {
...
+    CANT_GET_EXECUTABLE_NAME_FROM_PID(83),
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java]: Replace CANT with CANNOT

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java [hunk 3]
```diff
@@ -175,8 +175,12 @@ public enum ExitCode {
...
+    WRONGPATH_OF_EXECUTABLE(84);
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java]: Replace WRONGPATH with WRONG_PATH

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.c [hunk 4]
```diff
@@ -337,6 +337,16 @@ const char *get_error_message(const int error_code) {
         return "runC run failed";
       case ERROR_RUNC_REAP_LAYER_MOUNTS_FAILED:
         return "runC reap layer mounts failed";
+      case CANT_GET_EXECUTABLE_NAME_FROM_READLINK:
+        return "Can't get executable name from readlink";
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.c]: Replace Can't with Cannot

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.c [hunk 5]
```diff
@@ -337,6 +337,16 @@ const char *get_error_message(const int error_code) {
...
+      case TOO_LONG_EXECUTOR_PATH:
+        return "Too long executor path";
+      case CANT_GET_EXECUTABLE_NAME_FROM_KERNEL:
+        return "Can't get executable name from kernel";
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.c]: Replace Can't with Cannot

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.c [hunk 6]
```diff
@@ -337,6 +337,16 @@ const char *get_error_message(const int error_code) {
...
+      case CANT_GET_EXECUTABLE_NAME_FROM_PID:
+        return "Can't get executable name from pid";
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.c]: Replace Can't with Cannot

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.c [hunk 7]
```diff
@@ -337,6 +337,16 @@ const char *get_error_message(const int error_code) {
...
+      case WRONGPATH_OF_EXECUTABLE:
+        return "Wrongpath of executable";
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.c]: Replace "Wrongpath" with "Wrong path"

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.h [hunk 8]
```diff
@@ -104,7 +104,12 @@ enum errorcodes {
   ERROR_RUNC_SETUP_FAILED = 76,
   ERROR_RUNC_RUN_FAILED = 77,
   ERROR_RUNC_REAP_LAYER_MOUNTS_FAILED = 78,
-  ERROR_DOCKER_CONTAINER_EXEC_FAILED = 79
+  ERROR_DOCKER_CONTAINER_EXEC_FAILED = 79,
+  CANT_GET_EXECUTABLE_NAME_FROM_READLINK = 80,
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.h]: Replace CANT with CANNOT

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.h [hunk 9]
```diff
@@ -104,7 +104,12 @@ enum errorcodes {
...
+  TOO_LONG_EXECUTOR_PATH = 81,
+  CANT_GET_EXECUTABLE_NAME_FROM_KERNEL = 82,
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.h]: Replace CANT with CANNOT

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.h [hunk 10]
```diff
@@ -104,7 +104,12 @@ enum errorcodes {
...
+  CANT_GET_EXECUTABLE_NAME_FROM_PID = 83,
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.h]: Replace CANT with CANNOT

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.h [hunk 11]
```diff
@@ -104,7 +104,12 @@ enum errorcodes {
...
+  WRONGPATH_OF_EXECUTABLE = 84
```
Comment (szilard-nemeth) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/util.h]: Replace WRONGPATH with WRONG_PATH

**Review by szilard-nemeth — CHANGES_REQUESTED** [review]

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 58s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  42m 24s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   1m 34s |  |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   0m 48s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 55s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 50s |  |  trunk passed  |
| +1 :green_heart: |  spotbugs  |   1m 51s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  23m 44s |  |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  24m  6s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 40s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 19s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   1m 19s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   1m 19s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 19s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 34s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 35s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  23m 56s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  23m 48s |  |  hadoop-yarn-server-nodemanager in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 40s |  |  The patch does not generate ASF License warnings.  |
|  |   | 126m 35s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4949 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets cc golang |
| uname | Linux eada834c321b 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 5067667111fbd7263ad000d43895e3ff1c6513b6 |
| Default Java | Red Hat, Inc.-1.8.0_352-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/5/testReport/ |
| Max. process+thread count | 601 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4949/5/console |
| versions | git=2.9.5 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by szilard-nemeth** [comment]
Thanks @susheel-gupta for working on this.
Latest patch LGTM, committed to trunk.

Thanks @K0K0V0K for the review.

**Pull request closed without merging**